packages/rollup-core/src/app/data/data-service.ts

```diff
@@ -346,10 +346,32 @@ export class DefaultDataService implements DataService {
    * @inheritDoc
    */
   public async updateBlockToProcessed(blockHash: string): Promise<void> {
-    return this.rdb.execute(`
-    UPDATE l1_block 
-    SET processed = TRUE 
-    WHERE block_hash = '${add0x(blockHash)}'`)
+    return this.rdb.execute(
+      `UPDATE l1_block 
+        SET processed = TRUE 
+        WHERE block_hash = '${add0x(blockHash)}'`
+    )
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public async getL1RollupStateRootCount(): Promise<number> {
```

Comment:
Should we specify `getFinalizedRollupStateRootCount`?

Reply:
Finalized is pretty overloaded, so probably don't want to go with that, but it may be useful to indicate that this will not include roots that were removed due to fraud if that's helpful.

Thinking about it more, how will we fix fraud? As it stands, it cannot go through standard state root appending because the index logic will not work (and likewise the log parsing logic will not work). I think we may just want to remove the `WHERE removed = false` part of this, as a past fraudulent state root does not affect the index of a current state root.

Reply:
@ben-chain thoughts?

Reply:
After talking with Ben, realized that we're talking about two different things. Noting here for transparency:

Nothing goes into our DB from L1 unless it is "final" so clarifying that it is finalized shouldn't matter.